app/src/main/contrib/reducers/responsibilitiesReducer.ts

```diff
@@ -0,0 +1,24 @@
+import {ResponsibilitiesAction, ResponsibilitiesTypes} from "../actionTypes/ResponsibilitiesTypes";
+import {ExtendedResponsibility, IExtendedResponsibilitySet} from "../models/ResponsibilitySet";
+
+export interface ResponsibilitiesState {
+    set: IExtendedResponsibilitySet;
+    currentResponsibility: ExtendedResponsibility;
+}
+
+export const initialState: ResponsibilitiesState = {
+    set: null,
+    currentResponsibility: null,
+};
+
+export const responsibilitiesReducer = (state = initialState, action: ResponsibilitiesAction) => {
+    switch (action.type) {
+        case ResponsibilitiesTypes.SET_RESPONSIBILITIES:
+            return {...state, set: action.data };
+        case ResponsibilitiesTypes.SET_CURRENT_RESPONSIBILITY_SET:
```

Comment:
The action type here is called `SET_CURRENT_RESPONSIBILITY_SET` but actually the logic is setting the current responsibility, not the current responsibility set! I think the logic is right, and the action type should be `SET_CURRENT_RESPONSIBILITY`